Bottom: 4efdda093b660a293102621519e52c7da37c2789
Top:    e03200d6c3a80aa33cf7799e15c618e3c8d8d10a
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-04 20:07:09 +0000

Fixup some bugs from the added soldermask layers

(TODO: Finish this fixup and merge it down into the relevant patches)


---

diff --git a/src/find.c b/src/find.c
index 97f92a2..c402568 100644
--- a/src/find.c
+++ b/src/find.c
@@ -776,6 +776,8 @@ LookupLOConnectionsToLOList (int flag, bool AndRats)
   Cardinal i, group, layer, ratposition,
     lineposition[MAX_LAYER],
     polyposition[MAX_LAYER], arcposition[MAX_LAYER], padposition[2];
+  Cardinal top_group = GetLayerGroupNumberBySide (TOP_SIDE);
+  Cardinal bottom_group = GetLayerGroupNumberBySide (BOTTOM_SIDE);
 
   /* copy the current LO list positions; the original data is changed
    * by 'LookupPVConnectionsToLOList()' which has to check the same
@@ -848,22 +850,18 @@ LookupLOConnectionsToLOList (int flag, bool AndRats)
                         (POLYGONLIST_ENTRY (layer, *position), group, flag, AndRats))
                       return (true);
                 }
-              else
-                {
-                  /* try all new pads */
-                  layer -= max_copper_layer;
-                  if (layer > 1)
-                    {
-                      Message (_("bad layer number %d max_copper_layer=%d in find.c\n"),
-                               layer, max_copper_layer);
-                      return false;
-                    }
-                  position = &padposition[layer];
-                  for (; *position < PadList[layer].Number; (*position)++)
-                    if (LookupLOConnectionsToPad
-                        (PADLIST_ENTRY (layer, *position), group, flag, AndRats))
-                      return (true);
-                }
+            }
+
+          /* try all new pads */
+          if (group == top_group || group == bottom_group)
+            {
+              int side = (group == top_group) ? TOP_SIDE : BOTTOM_SIDE;
+
+              position = &padposition[side];
+              for (; *position < PadList[side].Number; (*position)++)
+                if (LookupLOConnectionsToPad
+                    (PADLIST_ENTRY (side, *position), group, flag, AndRats))
+                  return (true);
             }
         }
 
diff --git a/src/misc.c b/src/misc.c
index 4f381ae..31021c9 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -2015,6 +2015,7 @@ MoveLayerToGroup (int layer, int group)
 {
   int prev, i, j;
 
+#warning BUG NEXT LINE?
   if (layer < 0 || layer > max_copper_layer + 1)
     return -1;
   prev = GetLayerGroupNumberByNumber (layer);
diff --git a/src/move.c b/src/move.c
index b9a180d..5f0cfb3 100644
--- a/src/move.c
+++ b/src/move.c
@@ -928,12 +928,14 @@ MoveLayer (int old_index, int new_index)
   AddLayerChangeToUndoList (old_index, new_index);
   IncrementUndoSerialNumber ();
 
+#warning BUG NEXT LINE?
   if (old_index < -1 || old_index >= max_copper_layer)
     {
       Message ("Invalid old layer %d for move: must be -1..%d\n",
 	       old_index, max_copper_layer - 1);
       return 1;
     }
+#warning UNNECESSARY TEST AGAINST MAX_LAYER, max_copper_layer is same or smaller
   if (new_index < -1 || new_index > max_copper_layer || new_index >= MAX_LAYER)
     {
       Message ("Invalid new layer %d for move: must be -1..%d\n",
